test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,226 @@
+		if err != nil {
+			t.Logf("GET %s failed: %v, retrying...", r.Spec.Host, err)
+			return false, nil
+		}
+		if code != http.StatusOK {
+			t.Logf("GET %s failed: status %v, expected %v, retrying...", r.Spec.Host, code, http.StatusOK)
+			return false, nil // retry on 503 as pod/service may not be ready
+		}
+
+		if header.Get("Content-Encoding") != "gzip" {
+			t.Fatalf("compression is not enabled but should be for %s", r.Name)
+		}
+		t.Logf("compression is properly enabled for %s", r.Name)
+
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to get results on GET: %v", err)
+	}
+}
+
+// Get HTTP Headers.  If the request includes a Header of "Accept-Encoding: gzip" and the route
+// mimeType matches the ingress setting, we should see "Content-Encoding: gzip" in the response.
+func getHttp(client *http.Client, route *routev1.Route, addHeader bool) (http.Header, int, error) {
+	// Create the HTTPS request
+	request, err := http.NewRequest("GET", "https://"+route.Spec.Host, nil)
+	if err != nil {
+		return nil, -1, fmt.Errorf("New request failed: %v", err)
+	}
+
+	// Give the instruction to compress
+	if addHeader {
+		request.Header.Add("Accept-Encoding", "gzip")
+	}
+
+	response, err := client.Do(request)
+	if err != nil {
+		return response.Header, response.StatusCode, fmt.Errorf("GET %s failed: %v", route.Spec.Host, err)
+	}
+	// Close response body
+	defer response.Body.Close()
+
+	return response.Header, response.StatusCode, nil
+}
+
+// Update the spec.httpCompressionPolicy with different MIME types
+func testCompressionPolicy(t *testing.T, icName types.NamespacedName, domain string, compressionPolicy operatorv1.HTTPCompressionPolicy) error {
+	ic := newPrivateController(icName, domain)
+	ic.Spec.HTTPCompression = compressionPolicy
+
+	// Create a new Ingress Controller
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		return err
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	conditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+	}
+	// Wait for it to become available
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, conditions...); err != nil {
+		t.Logf("failed to observe expected conditions: %v", err)
+		return err
+	}
+
+	p := ic.Spec.HTTPCompression
+	var found bool
+	for _, given := range compressionPolicy.MimeTypes {
+		found = false
+		for _, stored := range p.MimeTypes {
+			if given == stored {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Fatalf("did not find mimeType: %s, found: %v\n", given, p.MimeTypes)
+		}
+	}
+	t.Logf("successfully created %s with MIME types: %v\n", ic.Name, ingresscontroller.GetMIMETypes(p.MimeTypes))
+	return nil
```

Comment:
What's the point of this logic?  `testCompressionPolicy` assigns `ic.Spec.HTTPCompression = compressionPolicy` above, so it seems like this logic is just checking that the value that was assigned to `ic.Spec.HTTPCompression` matches the value that is in `ic.Spec.HTTPCompression`, which should be the same value anyway.  If you want to verify that the MIME types are set on the router deployment, you could use `waitForDeploymentEnvVar`.

Reply:
Thanks.  I'll use waitForDeploymentEnvVar, I hadn't considered that was possible.